mutation_partition_view.cc

```diff
@@ -266,7 +266,8 @@ future<> mutation_partition_view::do_accept_gently(const column_mapping& cm, Vis
 
     for (auto&& cr : mpv.rows()) {
         auto t = row_tombstone(cr.deleted_at(), shadowable_tombstone(cr.shadowable_deleted_at()));
-        visitor.accept_row(position_in_partition_view::for_key(cr.key()), t, read_row_marker(cr.marker()), is_dummy::no, is_continuous::yes);
+        auto key = cr.key();
+        visitor.accept_row(position_in_partition_view::for_key(key), t, read_row_marker(cr.marker()), is_dummy::no, is_continuous::yes);
```

Comment:
How about a comment here explaining the reason? So someone doesn't optimize it back into the original form.

Reply:
Will do.

And I'll also look into a suggestion from @xemul to delete the position_in_partition_view(clustering_key_prefix&&) constructor(s), to prevent this from happening elsewhere.

Reply:
Wait, is this a miscompile, or a genuine bug?

Reply:
Frankly, I'm not sure.
After discussing with @denesb and @xemul we came to the conclusion that c++ does extend the life time of a temporary object when it is pass to a function as a const&, but here `position_in_partition_view::for_key` generates a view that keeps a reference to the temporary, so the compiler doesn't necessarily know about it.

In sequential code, like https://godbolt.org/z/dadW6s674, it does extend the life time of the temporary until the call stack resolves, but for some reason this breaks on arm with optimization (dev/release modes) when the coroutine yields.
But note that the problem is manifested only after resuming the coroutine, where a new key is supposed to be created, so I'm not sure exactly what's going on.

Reply:
If you change maybe_yield to yield below, does it break on x86? Maybe the difference is more yielding on arm.

Reply:
> If you change maybe_yield to yield below, does it break on x86? Maybe the difference is more yielding on arm.

No, it doesn't reproduce on x86 with unconditional `co_await yield()`.